pkg/operator/controllers/monitoring/monitoring.go

```diff
@@ -0,0 +1,193 @@
+// The ConfigMap must exist before setting up the reconciler
+func EnsureConfigMap(ctx context.Context, kubernetescli kubernetes.Interface) (*v1.ConfigMap, error) {
+	var cm *v1.ConfigMap
+	name := monitoringName.Name
+	ns := monitoringName.Namespace
+	err := retry.RetryOnConflict(retry.DefaultRetry, func() error {
+		var err error
+		cm, err = kubernetescli.CoreV1().ConfigMaps(ns).Get(ctx, name, metav1.GetOptions{})
+		if err != nil {
+			if _, ok := err.(*errors.StatusError); ok {
+				cm, err = kubernetescli.CoreV1().ConfigMaps(ns).Create(ctx, &v1.ConfigMap{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      name,
+						Namespace: ns,
+					},
+					Data: map[string]string{
+						"config.yaml": defaultConfig,
+					},
+				}, metav1.CreateOptions{})
+				return err
+			}
+		}
+		return err
+	})
+	return cm, err
+}
+
+func (r *Reconciler) reconcileMonitoringConfig(ctx context.Context) error {
+	cm, err := EnsureConfigMap(ctx, r.kubernetescli)
+	if err != nil {
+		return err
+	}
+
+	return retry.RetryOnConflict(retry.DefaultRetry, func() error {
+		if cm.Data == nil {
+			cm.Data = map[string]string{
+				"config.yaml": "",
+			}
+		}
+
+		configDataJSON, err := yaml.YAMLToJSON([]byte(cm.Data["config.yaml"]))
+		if err != nil {
+			return err
+		}
+
+		var configData config
+		err = codec.NewDecoderBytes(configDataJSON, &codec.JsonHandle{}).Decode(&configData)
+		if err != nil {
+			return err
+		}
+
+		changed := false
+		if configData.PrometheusK8s.Retention != "15d" {
+			configData.PrometheusK8s.Retention = "15d"
+			changed = true
+		}
+
+		if configData.PrometheusK8s.VolumeClaimTemplate.Spec.Resources.Requests.Storage != "100Gi" {
+			configData.PrometheusK8s.VolumeClaimTemplate.Spec.Resources.Requests.Storage = "100Gi"
+			changed = true
+		}
+
+		if !changed {
+			return nil
+		}
+
+		var b []byte
+		err = codec.NewEncoderBytes(&b, &codec.JsonHandle{}).Encode(configData)
+		if err != nil {
+			return err
+		}
+
+		cmYaml, err := yaml.JSONToYAML(b)
+		if err != nil {
+			return err
+		}
+		cm.Data["config.yaml"] = string(cmYaml)
+
+		_, err = r.kubernetescli.CoreV1().ConfigMaps(monitoringName.Namespace).Update(ctx, cm, metav1.UpdateOptions{})
+		return err
+	})
+}
+
+func triggerMonitoringReconcile(cm *corev1.ConfigMap) bool {
+	return cm.Name == monitoringName.Name && cm.Namespace == monitoringName.Namespace
+}
+
+// SetupWithManager setup the mananger
+func (r *Reconciler) SetupWithManager(mgr ctrl.Manager) error {
+	r.log.Info("starting starting cluster monitoring controller")
+
+	isMonitoringManager := predicate.Funcs{
+		UpdateFunc: func(e event.UpdateEvent) bool {
+			cm, ok := e.ObjectOld.(*corev1.ConfigMap)
+			return ok && triggerMonitoringReconcile(cm)
+		},
+		CreateFunc: func(e event.CreateEvent) bool {
+			cm, ok := e.Object.(*corev1.ConfigMap)
+			return ok && triggerMonitoringReconcile(cm)
+		},
+	}
+
+	return ctrl.NewControllerManagedBy(mgr).
+		For(&v1.ConfigMap{}).
```

Comment:
I think that doing what the pull secret controller here is useful:
```golang
		For(&arov1alpha1.Cluster{}).
		// https://github.com/kubernetes-sigs/controller-runtime/issues/1173
		// equivalent to For(&v1.ConfigMap{})., but can't call For multiple times on one builder
		Watches(&source.Kind{Type: &v1.ConfigMap{}}, &handler.EnqueueRequestForObject{}).
```

This removes the requirement for EnsureConfigMap because we will get notified on the Cluster resource as well as the ConfigMap in question.